regparser/tree/xml_parser/paragraph_processor.py

```diff
@@ -147,21 +160,56 @@ class TableMatcher(object):
     def matches(self, xml):
         return xml.tag == 'GPOTABLE'
 
-    def derive_nodes(self, xml):
+    def derive_nodes(self, xml, processor=None):
         return [Node(table_xml_to_plaintext(xml), label=[mtypes.MARKERLESS],
                      source_xml=xml)]
 
 
+class HeaderMatcher(object):
+    def matches(self, xml):
+        return xml.tag == "HD"
+
+    def derive_nodes(self, xml, processor=None):
+        # This should match HD elements only at lower levels, and for now we'll
+        # just put them into the titles
+        return [Node(text='', title=tree_utils.get_node_text(xml).strip(),
+                     label=[mtypes.MARKERLESS])]
+
+
+class ExtractMatcher(object):
+    def matches(self, xml):
+        return xml.tag in ('EXTRACT')
+
+    def derive_nodes(self, xml, processor=None):
+        extract_node = Node(text=xml.text, node_type=u"extract",
+                            label=[mtypes.MARKERLESS])
+        child_nodes = processor.parse_nodes(xml)
+
+        if child_nodes:
+            markers = [node.label[0] for node in child_nodes]
+            depths = derive_depths(markers, processor.additional_constraints())
+            if not depths:
+                logging.error(
+                    "Could not determine paragraph depths:\n%s", markers)
+            depths = processor.select_depth(depths)
+            hierarchy = processor.build_hierarchy(extract_node,
+                                                  child_nodes, depths)
+            extract_node = hierarchy
+
+        return [extract_node]
```

Comment:
Good catch! In prior versions there were reasons not to do that but they all proved to be extraneous.